src/core/qgis.h

```diff
@@ -2088,6 +2088,19 @@ class CORE_EXPORT Qgis
     };
     Q_ENUM( JoinStyle )
 
+    /**
+     * Join styles for 3D buffers.
+     *
+     * \since QGIS 3.44
+     */
+    enum class JoinStyle3D SIP_MONKEYPATCH_SCOPEENUM_UNNEST( QgsGeometry, JoinStyle3D ) : int
+      {
+      Round SIP_MONKEYPATCH_COMPAT_NAME( JoinStyle3DRound ) = 1, //!< Smooth, rounded buffer around the input geometry
+      Flat SIP_MONKEYPATCH_COMPAT_NAME( JoinStyle3DFlat ), //!< Flat ends and constant width along the linestring
+      CylSphere SIP_MONKEYPATCH_COMPAT_NAME( JoinStyle3DCylSphere ), //!< Cylinders along the linestring segments with spheres at the vertices
+    };
+    Q_ENUM( JoinStyle3D )
```

Comment:
remove monkey patching compat code for new enums, avoid abbreviations

```suggestion
    /**
     * Join style for 3D buffers.
     *
     * \since QGIS 4.0
     */
    enum class JoinStyle3D : int
      {
      Round = 1, //!< Smooth, rounded buffer around the input geometry
      Flat, //!< Flat ends and constant width along the linestring
      CylindersAndSpheres, //!< Cylinders along the linestring segments with spheres at the vertices
    };
    Q_ENUM( JoinStyle3D )
```

Reply:
done